### ISSUE TITLE
avoid-leaking-state problem

### ISSUE THREAD

**Issue by jcbvm** [ember-cli/eslint-plugin-ember#203]
From the docs, the solution for defining object properties as list or object is to assign its value in the `init` method. The problem is however that when you do this for a component, the given property values in the template will be overridden by the values in the `init` method.

Is there any other solution for this case? I want to have a default value defined for properties which are not passed in via a template.

**Comment by rwjblue** [comment]
Can you give a more concrete example?

**Comment by jcbvm** [comment]
Example below will illustrate the issue, this is a more simple case instead of component:

```javascript
const Test = EmberObject.extend({
    test: null,

    init() {
        this._super(...arguments);
        this.test = [];
    }
});

// below will not set test to ['en'] because it gets overridden by init method above
Test.create({
    test: ['en']
});
```

**Comment by rwjblue** [comment]
Right, you have to use a conditional.  One example would be:

```js
const Test = EmberObject.extend({
    init() {
        this._super(...arguments);
        this.test = this.test || [];
    }
});
```

Another way (if falsey values are allowed):

```js
const Test = EmberObject.extend({
  init() {
        this._super(...arguments);
        this.test = 'test' in this ? this.test : [];
    }
});
```

**Comment by jcbvm** [comment]
That will do the job indeed. 
Should this be mentioned in the docs?

**Comment by rwjblue** [comment]
Yep, sounds good to have the documentation cover this scenario, would love a PR...
